### PR TITLE
Fix numerical bug that causes NaNs in DeepCSV output

### PR DESCRIPTION
**Pull request by mondalspandan**
Running b-tagging sequence in 94X and later stores NaNs in DeepCSV outputs for some jets. The problem can be traced to NaNs being fed as inputs to the NN. Specifically, the inputs trackPtRel and trackPtRatio are are returned as NaNs for certain tracks whenever the square of a quantity is evaluated as negative due to floating point errors. This is fixed by setting the quantity to zero whenever its square is evaluated as negative (=> the square is small enough to be affected by floating point errors).

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoBTag/SecondaryVertex/interface/CombinedSVComputer.h [hunk 1]
```diff
@@ -221,10 +221,13 @@ void CombinedSVComputer::fillCommonVariables(reco::TaggingVariableList & vars, r
                 vars.insert(btau::trackMomentum, trackMag, true);
                 vars.insert(btau::trackEta, trackMom.Eta(), true);
 
-                vars.insert(btau::trackPtRel, VectorUtil::Perp(trackMom, jetDir), true);
+                // check for erroneous Perp^2 values before evaluating Perp (fix for DeepCSV NaN outputs)
+                double Perp_trackMom_jetDir = VectorUtil::Perp2(trackMom, jetDir) > 0. ? VectorUtil::Perp(trackMom, jetDir) : 0.;
```
Comment (perrotta) [inline RecoBTag/SecondaryVertex/interface/CombinedSVComputer.h]: Use lower case initials for the local variables:  ```perp_trackMom_jetDir```

A (tiny) optimization could be (compute the scalar product only once):
```
double perp_trackMom_jetDir = VectorUtil::Perp2(trackMom, jetDir);
perp_trackMom_jetDir = (perp_trackMom_jetDir > 0. ? std::sqrt(perp_trackMom_jetDir ) : 0. ); 
```

Comment (mondalspandan) [reply RecoBTag/SecondaryVertex/interface/CombinedSVComputer.h]: @perrotta Thanks! Updated.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Add protection for the possible square roots of negative values
- Jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**